sonar-core/src/main/java/org/sonar/core/platform/ComponentContainer.java

```diff
@@ -260,7 +261,16 @@ public void start(Object component) {
         profiler.stopTrace(component.getClass().getCanonicalName() + " started");
       }
     };
-    return new DefaultPicoContainer(new OptInCaching(), lifecycleStrategy, null);
+    return new DefaultPicoContainer(new OptInCaching(), lifecycleStrategy, null) {
+      @Override
+      public Object getComponent(final Object componentKeyOrType, final Class<? extends Annotation> annotation) {
+        try {
+          return super.getComponent(componentKeyOrType, annotation);
+        } catch (Throwable t) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1181)